discovery_engine_core/web-api/src/handlers.rs

```diff
@@ -52,7 +57,13 @@ pub(crate) async fn handle_user_interaction(
                     .log_positive_user_reaction(positive_cois, &document.embedding)
             })
             .await
-            .map_err(handle_user_state_op_error)?;
+            .map_err(|err| {
+                error!(
+                    "Error updating cois for user '{}' with document '{}': {:#?}",
+                    user_id, body.document_id, err
+                );
+                handle_user_state_op_error(err)
+            })?;
 
         Ok(StatusCode::OK)
     } else {
```

Comment:
maybe a `debug` in else block with the `user_id` and `body.document_id` that was not found if that doesn't show up already in log output from `warp::trace::request()`